lib/createExplorer.js

```diff
@@ -10,10 +10,10 @@ var funcRunner = require('./funcRunner');
 
 module.exports = function (options) {
   // When `options.sync` is `false` (default),
-// these cache Promises that resolve with results, not the results themselves.
+  // these cache Promises that resolve with results, not the results themselves.
   var fileCache = (options.cache) ? new Map() : null;
   var directoryCache = (options.cache) ? new Map() : null;
-  var transform = options.transform || (options.sync) ? identitySync : identityPromise;
+  var transform = options.transform || (options.sync ? identitySync : identityPromise);
```

Comment:
I'd suggest we clarify this syntax more to avoid misunderstandings in the future. I certainly missed this subtlety. How about:

```js
var defaultTransform = options.sync ? identitySync : identityPromise;
var transform = options.transform || defaultTransform;
```